config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java

```diff
@@ -96,7 +114,13 @@ private Entity buildEncassEntity(Bundle bundle, String name, Encass encass, Docu
         final Map<String, Object> properties = Optional.ofNullable(encass.getProperties()).orElse(new HashMap<>());
         properties.putIfAbsent(PALETTE_FOLDER, DEFAULT_PALETTE_FOLDER_LOCATION);
         buildAndAppendPropertiesElement(properties, document, encassAssertionElement);
-        return getEntityWithNameMapping(ENCAPSULATED_ASSERTION_TYPE, name, id, encassAssertionElement);
+        Entity entity = getEntityWithNameMapping(ENCAPSULATED_ASSERTION_TYPE, encassName, id, encassAssertionElement);
+        if (annotatedEncassEntity != null && annotatedEncassEntity.isReusableEntity()) {
```

Comment:
are we generating reusable entities for UN-annotated bundles?

Reply:
no,  annotatedEntity will be null in other case